admin/includes/functions/html_output.php

```diff
@@ -503,34 +503,49 @@ function zen_draw_date_selector($fieldname_prefix, $default_date='') {
     return $date_selector;
 }
 
-function addSearchKeywordForm($filename, $action) {
-    $keywords_products = (isset($_POST['keywords']) && zen_not_null($_POST['keywords'])) ? zen_db_input(zen_db_prepare_input($_POST['keywords'])) : '';
-    $fullAction = '';
-    if (!empty($action)) {
-        $fullAction = 'action=' . $action;
-    }
-    $form = '
+    /**
+     * intended to add a universal search for use with any pulldown that extends the
+     * abstract pulldown class.  returns a html string.
+     * @param string $filename
+     * @param string $action
+     * @param bool $includeForm
+     * @param array $extrafieldsArray
+     * @return string
+     */
+
+    function addSearchKeywordForm(string $filename = '', string $action = '', bool $includeForm = true, array $extrafieldsArray = [])
+    {
+        $keywords_products = (isset($_POST['keywords']) && zen_not_null($_POST['keywords'])) ? zen_db_input(zen_db_prepare_input($_POST['keywords'])) : '';
+        $form = '';
+        $endForm = '';
+        if ($includeForm) {
+            $form = zen_draw_form('keywords', $filename, 'action=' . $action, 'post', 'class="form-horizontal"');
```

Comment:
Some forms need no 'action'; why was the `$fullAction` (i.e. include action only if not 'empty') portion removed?

Reply:
no idea.  it makes sense to have it back in.